Lib/gftools/builder/__init__.py

```diff
@@ -0,0 +1,341 @@
+                    self.masters[src].append(Babelfont.open(src, master=master.name))
+            elif src.endswith("designspace"):
+                # XXX
+                continue
+            else:
+                self.masters[src] = [Babelfont.open(src)]
+
+    def fill_config_defaults(self):
+        if "familyName" not in self.config:
+            self.logger.info("Deriving family name (this takes a while)")
+            self.load_masters()
+            familynames = set([x.info.familyName for x in self.masters.values()])
+            if len(familynames) > 1:
+                raise ValueError(
+                    "Inconsistent family names in sources (%s). Set familyName in config instead"
+                    % familynames
+                )
+            self.config["familyName"] = list(familynames)[0]
+        if "outputDir" not in self.config:
+            self.config["outputDir"] = "../fonts/"
+        if "vfDir" not in self.config:
+            self.config["vfDir"] = self.config["outputDir"] + "/variable/"
+        if "ttDir" not in self.config:
+            self.config["ttDir"] = self.config["outputDir"] + "/ttf/"
+        if "otDir" not in self.config:
+            self.config["otDir"] = self.config["outputDir"] + "/otf/"
+        if "woffDir" not in self.config:
+            self.config["woffDir"] = self.config["outputDir"] + "/webfonts/"
+
+        if "buildVariable" not in self.config:
+            self.config["buildVariable"] = True
+        if "buildStatic" not in self.config:
+            self.config["buildStatic"] = True
+        if "buildWebfont" not in self.config:
+            self.config["buildWebfont"] = self.config["buildStatic"]
+        if "autohintTTF" not in self.config:
+            self.config["autohintTTF"] = True
+        if "logLevel" not in self.config:
+            self.config["logLevel"] = "INFO"
+
+    def build_variable(self):
+        self.mkdir(self.config["vfDir"], clean=True)
+        args = {"output": ["variable"], "family_name": self.config["familyName"]}
+        all_variables = {}
+        for source in self.config["sources"]:
+            if not source.endswith(".designspace") and not source.endswith("glyphs"):
+                continue
+            self.logger.info("Creating variable fonts from %s" % source)
+            sourcebase = os.path.splitext(os.path.basename(source))[0]
+            args["output_path"] = os.path.join(
+                self.config["vfDir"], sourcebase + "-VF.ttf",
+            )
+            output_files = self.run_fontmake(source, args)
+            newname = self.rename_variable(output_files[0])
+            self.post_process(newname)
+            all_variables[source] = newname
+        if all_variables:
+            self.gen_stat(all_variables)
+
+    def run_fontmake(self, source, args):
+        if "output_dir" in args:
+            original_output_dir = args["output_dir"]
+            tmpdir = tempfile.TemporaryDirectory()
```

Comment:
TemporaryDirectory can be used as a context manger e.g

```
with TemporaryDirectory() as tmpdir:
    ...
```

This approach provides automatic cleanup so L205 becomes redundant.

Reply:
I looked at that, but we only need a temporary directory when we're producing a directory full of stuff (as opposed to, say, an individual variable font). There's no problem making a temp directory we're not going use, but is it really worth it to save a line? :-)